frontend/packages/topology/src/elements/BaseEdge.ts

```diff
@@ -88,9 +105,9 @@ export default class BaseEdge<E extends EdgeModel = EdgeModel, D = any> extends
     } else if (this.endPoint) {
       referencePoint = this.endPoint;
     } else {
-      referencePoint = this.targetAnchor.getReferencePoint();
+      referencePoint = this.targetAnchor ? this.targetAnchor.getReferencePoint() : Point.EMPTY;
     }
-    return this.sourceAnchor.getLocation(referencePoint);
+    return this.sourceAnchor ? this.sourceAnchor.getLocation(referencePoint) : Point.EMPTY;
```

Comment:
We should always have an anchor. If not the topology will be broken with a Point at 0,0.
With the other suggested changes, we can omit this change.